src/core/connection.c

```diff
@@ -3915,12 +3912,17 @@ QuicConnRecvDecryptAndAuthenticate(
                 Packet->PacketNumber);
 
         } else if (Packet->KeyType == QUIC_PACKET_KEY_1_RTT &&
-            Packet->PacketNumber < PacketSpace->ReadKeyPhaseStartPacketNumber) {
+            Packet->SH->KeyPhase == PacketSpace->CurrentKeyPhase) {
             //
-            // If this packet is the current key phase, but has an earlier packet
-            // number than this key phase's start, update the key phase start.
+            // This packet is in the current key phase, so update the packet space
+            // endpoints.
             //
-            PacketSpace->ReadKeyPhaseStartPacketNumber = Packet->PacketNumber;
+            if (Packet->PacketNumber < PacketSpace->ReadKeyPhaseStartPacketNumber) {
+                PacketSpace->ReadKeyPhaseStartPacketNumber = Packet->PacketNumber;
+            }
+            if (Packet->PacketNumber > PacketSpace->ReadKeyPhaseEndPacketNumber) {
+                PacketSpace->ReadKeyPhaseEndPacketNumber = Packet->PacketNumber;
+            }
             QuicTraceLogConnVerbose(
```

Comment:
Only log this if `ReadKeyPhaseStartPacketNumber` is updated? Otherwise every single new packet we receive in the current key phase will trigger this, right?